assets/src/components/conversations/ConversationsProvider.tsx

```diff
@@ -168,6 +184,69 @@ export class ConversationsProvider extends React.Component<Props, State> {
       });
   };
 
+  handlePresenceState = (state: any) => {
+    this.setState({presence: state});
+  };
+
+  updatePresenceWithJoins = (joins: any, presence: any) => {
+    return Object.keys(joins).reduce((acc, key) => {
```

Comment:
This return is kinda hard to understand. I m not really sure whats going on here. Also unclear on what is metas. Can we add some comments or maybe change the variable metas.

Reply:
Maybe instead of calling them joins maybe called the NewCustomers or some sort of noun. having a verb as a variable is throwing me off a bit.

Reply:
sounds good, i'll add some comments to clarify and also add the type signatures 👍